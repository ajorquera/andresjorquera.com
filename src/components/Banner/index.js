import React from "react"
import Button from 'react-bootstrap/Button'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import ReactGA from 'react-ga';

import avatar from "../../images/jonathan.png"

export default class Banner extends React.Component {
	hireMe() {
		const chat = document.querySelector('.rsc-float-button');
		chat.click();
		ReactGA.event({
			action: 'click',
			category: 'lead',
			value: 1
		});
	}
	render() {
		const links = {
			instagram: 'https://www.instagram.com/andres.codes/',
			linkedin: 'https://www.linkedin.com/in/andres-jorquera-mujica/',
			github: 'https://github.com/ajorquera'
		};
	
		const email = 'jorquera.ad@gmail.com';
	
		return (
			<div className="custom-banner jumbotron jumbotron-fluid py-3 row">
				<div className="text-center container my-auto"><img className="img-fluid mx-auto d-block" src={avatar} alt="avatar" />
					<div className="mt-1 mt-md-5">
						<ul className="brands brands-inline d-none d-md-inline mr-4">
							<li><a href={links.instagram} target="_blank" rel="noopener noreferrer">
								<FontAwesomeIcon color="white" icon={['fab', 'instagram']} />
							</a></li>
							<li><a href={links.linkedin} target="_blank" rel="noopener noreferrer">
								<FontAwesomeIcon color="white" icon={['fab', 'linkedin']} />
							</a></li>
						</ul>
						<h1 className="text-white display-4 font-weight-bolder d-inline">Andr√©s Jorquera</h1>
						<ul className="list-inline brands brands-inline d-none d-md-inline ml-4">
							<li><a href={'mailto:' + email + '?subject=Website Contact'}>
								<FontAwesomeIcon color="white" icon="envelope" />		
							</a></li>
							<li><a href={links.github} target="_blank" rel="noopener noreferrer">
								<FontAwesomeIcon color="white" icon={['fab', 'github']} />
							</a></li>
						</ul>
					</div>
					<div className="carousel slide vertical mt-2 my-md-4" data-ride="carousel">
						<div className="carousel-inner">
							<div className="item active">
								<h3> I <span className="d-md-block">
									<FontAwesomeIcon color="red" icon="heart" />
								</span>&nbsp;programming</h3>
							</div>
						</div>
					</div>
	
					<style type="text/css">
						{`
							.btn-outline-custom {
								border: 1px solid #FFFFFF;
								border-bottom: 3px solid #FFFFFF;
								background-color: transparent;
							}
	
							.btn-outline-custom:hover {
								border: 1px solid #16A085;
								border-bottom: 3px solid #16A085;
								color: #16A085;
							}
						`}
					</style>
	
					<Button size="lg" variant="outline-custom" className="my-1" href="#about">
						Know me better
					</Button>
					<Button onClick={this.hireMe.bind(this)} size="lg" variant="outline-custom" className="my-2" href="#contact" id="hire-me-button" role="button">
						<FontAwesomeIcon icon="bolt" />
						&nbsp;Hire me
					</Button>
				</div>
			</div>
		)
	}
};
