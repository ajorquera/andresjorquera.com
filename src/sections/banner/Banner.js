import React from "react"
import Button from 'react-bootstrap/Button'
import loadable from '@loadable/component'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import ReactGA from 'react-ga';
import BannerLink from '@components/BannerLink'
import {FormattedMessage} from 'react-intl';


import avatar from "@images/jonathan.png"
const TextSlide = loadable(() => import('@components/TextSlide'))

//NOTE: fix for when changing img size. 
const hotfixStyle = {
	'overflow': 'hidden',
    'maxHeight': '65px'
}

let slides = [
	[{text: (<FormattedMessage id="i" />)}, {icon:'heart', color: 'red'}, {text: (<FormattedMessage id="creating apps" />)}],
	[{text: (<FormattedMessage id="i'm here to" />)}, {icon:'handshake'}, {text: (<FormattedMessage id="together" />)}],
	[{text: (<FormattedMessage id="i" />)}, {icon:'lightbulb', color:'yellow'}, {text: (<FormattedMessage id="different" />)}],
	[{text: (<FormattedMessage id="i" />)}, {icon:'wrench', color: 'gray', inline: true}, {text: (<FormattedMessage id="with" />), inline: true}, {text: (<FormattedMessage id="transparency" />)}]
];

export default class Banner extends React.Component {
	hireMe() {
		const chat = document.querySelector('.rsc-float-button');
		chat.click();
		ReactGA.event({
			action: 'click',
			category: 'lead',
			value: 1
		});
	}
	render() {
		const links = {
			instagram: 'https://www.instagram.com/andres.codes/',
			linkedin: 'https://www.linkedin.com/in/andres-jorquera-mujica/',
			github: 'https://github.com/ajorquera'
		};
	
		const email = this.props.email;
		const name = this.props.name;
		return (
			<div className="custom-banner jumbotron jumbotron-fluid py-3 row">
				<div className="text-center container my-auto"><img className="img-fluid mx-auto d-block" src={avatar} alt="avatar" />
					<div className="mt-1 mt-md-5" style={hotfixStyle}>
						<span className="d-none d-md-inline mr-4">
							<BannerLink className="mx-1" link={links.instagram} icon='instagram' />
							<BannerLink className="mx-1" link={links.linkedin} icon='linkedin' />
						</span>
						<h1 className="text-white display-4 font-weight-bolder d-inline">{name}</h1>
						<span className="d-none d-md-inline ml-4">
							<BannerLink className="mx-1" link={'mailto:' + email + '?subject=Website Contact'} icon='envelope' /> 
							<BannerLink className="mx-1" link={links.github} icon='github' /> 	
						</span>
					</div>
					
					<TextSlide slides={slides} />
	
					<style type="text/css">
						{`
							.btn-outline-custom {
								border: 1px solid #FFFFFF;
								border-bottom: 3px solid #FFFFFF;
								background-color: transparent;
							}

							.banner-icon {
								color: #93c7ef;
								transition: color .4s ;
							}

							.btn-outline-custom:hover .banner-icon {
								color: #16A085
							}
	
							.btn-outline-custom:hover {
								border: 1px solid #16A085;
								border-bottom: 3px solid #16A085;
								color: #16A085;
							}
						`}
					</style>
	
					<Button onClick={this.hireMe.bind(this)} size="lg" variant="outline-custom" className="my-2" href="#contact" id="hire-me-button" role="button">
						<FormattedMessage id="Speak to" />
						&nbsp;
						<FontAwesomeIcon className="banner-icon" icon="robot" color="#93c7ef" />
					</Button>
				</div>
			</div>
		)
	}
};
