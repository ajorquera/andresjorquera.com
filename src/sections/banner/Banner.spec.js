import React from "react"
import renderer from "react-test-renderer"
import {IntlProvider} from "react-intl";

import Banner from ".";


describe("banner", () => {
  beforeAll(() => {
    console.error = jest.fn()
  });

  afterAll(() => {
    jest.restoreAllMocks()
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale="es">
          <Banner name="Andres Jorquera" email="jorquera.ad@gmail.com" />
          </IntlProvider>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})