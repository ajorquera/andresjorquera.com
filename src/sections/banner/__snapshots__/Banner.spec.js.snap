// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner renders correctly 1`] = `
<div
  className="custom-banner jumbotron jumbotron-fluid py-3 row"
>
  <div
    className="text-center container my-auto"
  >
    <img
      alt="avatar"
      className="img-fluid mx-auto d-block"
      src="test-file-stub"
    />
    <div
      className="mt-1 mt-md-5"
      style={
        Object {
          "maxHeight": "65px",
          "overflow": "hidden",
        }
      }
    >
      <span
        className="d-none d-md-inline mr-4"
      >
        <a
          className="mx-1 sc-bdVaJa lnheUV"
          href="https://www.instagram.com/andres.codes/"
          rel="noopener noreferrer"
          target="_blank"
        />
        <a
          className="mx-1 sc-bdVaJa lnheUV"
          href="https://www.linkedin.com/in/andres-jorquera-mujica/"
          rel="noopener noreferrer"
          target="_blank"
        />
      </span>
      <h1
        className="text-white display-4 font-weight-bolder d-inline"
      >
        Andres Jorquera
      </h1>
      <span
        className="d-none d-md-inline ml-4"
      >
        <a
          className="mx-1 sc-bdVaJa lnheUV"
          href="mailto:jorquera.ad@gmail.com?subject=Website Contact"
          rel="noopener noreferrer"
          target="_blank"
        />
        <a
          className="mx-1 sc-bdVaJa lnheUV"
          href="https://github.com/ajorquera"
          rel="noopener noreferrer"
          target="_blank"
        />
      </span>
    </div>
    <style
      type="text/css"
    >
      
							.btn-outline-custom {
								border: 1px solid #FFFFFF;
								border-bottom: 3px solid #FFFFFF;
								background-color: transparent;
							}

							.banner-icon {
								color: #93c7ef;
								transition: color .4s ;
							}

							.btn-outline-custom:hover .banner-icon {
								color: #16A085
							}
	
							.btn-outline-custom:hover {
								border: 1px solid #16A085;
								border-bottom: 3px solid #16A085;
								color: #16A085;
							}
						
    </style>
    <a
      className="my-2 btn btn-outline-custom btn-lg"
      href="#contact"
      id="hire-me-button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span>
        Speak to
      </span>
      Â 
    </a>
  </div>
</div>
`;
